source/CardGames.py

```diff
@@ -154,22 +154,191 @@ def resetDeck(self):
+        index += 1
+    
+    self.cards = deck
+    self.size = len(deck)
+    self.cardBack = cardBack
+    self.discarded = []
 
+  def reset(self):
+    self.cards += self.discarded
+    self.discarded = []
+    self.size = len(self.cards)
 
+  def shuffle(self):
+    random.shuffle(self.cards)
 
+  def getCard(self):
+    card = self.cards.pop()
+    self.size -= 1
+    self.discarded.append(card)
+    return card
 
+def getCard( suit, value):
+  deck = Deck()
+  my_card = Card( suit.capitalize(), value, None, None)
+  for card in deck.cards:
+    if card == my_card:
+      return card
+  return None
 
+class Player:
+  def __init__(self, name, money: int = 0):
+    self.name = name
+    self.hand = []
+    self.knownCards = []
+    self.money = money
 
+  def addMoney(self, amount: int):
+    self.money += amount
+    return self.money
 
+  def makeBet(self, amount: int):
+    if amount > self.money:
+      print("%s does not have enough money to make this bet." % self.name)
+      return self.money
+    self.money -= amount
+    return self.money
 
+  def addCard(self, card: Card, isKnown: bool = True):
+    self.hand.append(card)
+    if isKnown:
+      self.knownCards.append(True)
+    else:
+      self.knownCards.append(False)
 
+  def setHand(self, cards: "list[Card]", isKnown: bool = False):
+    self.hand = cards
+    self.knownCards = [isKnown for _ in self.hand]
 
+  def showHand(self, printShort: bool = False):
+    for idx in range(6):
+      for i, card in enumerate(self.hand):
+        if printShort and i < len(self.hand)-1:
+          image = card.shortImage[idx]  if self.knownCards[i] else card.cardBack[idx]
+          print(image, end="")
+        else:
+          image = card.image[idx] if self.knownCards[i] else card.cardBack[idx]
+          print(image, end="")
+      print()
 
+  def clearHand(self):
+    self.hand = []
+    self.knownCards = []
 
+class Dealer:
+  def __init__(self, deck: Deck):
+    self.deck = deck
+    self.deck.shuffle()
 
+  def printCards(self, cards: "list[Card]", showFront: bool, printShort: bool = True):
+    for idx in range(6):
+      for i, card in enumerate(cards):
+        if printShort and i < len(cards)-1:
+          image = card.shortImage[idx] if showFront else card.cardBack[idx]
+          print(image, end="")
+        else:
+          image = card.image[idx] if showFront else card.cardBack[idx]
+          print(image, end="")
+      print()
 
+  def dealCards(self, numCards: int, players: "list[Player]"):
+    if numCards * len(players) > self.deck.size:
+      return False
+    for player in players:
+      for _ in range(numCards):
+        player.addCard(self.deck.getCard())
+    return True
 
+  def resetDeck(self):
+    self.deck.reset()
+    self.deck.shuffle()
+    
+def character_selection():
+  Player1 = input("Player 1, please enter your name: ")
+  Player2 = input("Player 2, please enter your name: ")
+  Player1Role = ""
```

Comment:
Looks great. Since we're just doing one Player class, you can probably just assign the role to a variable in the class like Player1.setRole('fugitive').